06-building-decays.md

```diff
@@ -149,8 +186,26 @@ from Configurables import DaVinci
 DaVinci().UserAlgorithms += [dstar_seq.sequence()]
 ```
 
+
+> ## Debugging your selection chain {.callout}
+> The `PhysSelPython.Wrappers` offers a very useful utility for debugging your selection chains, called `PrintSelection`.
+> It gets a `Selection` as input and it can be used the same way, except it will print the decay tree everytime making use of the `PrintDecayTree` algorithm which was discussed in the [Exploring a DST](05-interactive-dst.html) lesson.
+>
+> For more complex debugging, one can setup `DaVinci` with `graphviz
+> ```shell
+> SetupDaVinci --use "graphviz v* LCG_Interfaces"
+> ```
+>
+> And add a nice graphical representation of the selection sequence
+> ```python
+>from SelPy.graph import graph
```

Comment:
It is.

On Wed, Jun 3, 2015 at 10:13 AM, Tim Head <notifications@github.com> wrote:

> In 06-building-decays.md
> <https://github.com/lhcb/first-analysis-steps/pull/90#discussion_r31601822>
> :
>
> > @@ -149,8 +186,26 @@ from Configurables import DaVinci
> >  DaVinci().UserAlgorithms += [dstar_seq.sequence()]
> >  ```
> >
> > +
> > +> ## Debugging your selection chain {.callout}
> > +> The `PhysSelPython.Wrappers` offers a very useful utility for debugging your selection chains, called `PrintSelection`.
> > +> It gets a `Selection` as input and it can be used the same way, except it will print the decay tree everytime making use of the `PrintDecayTree` algorithm which was discussed in the [Exploring a DST](05-interactive-dst.html) lesson.
> > +>
> > +> For more complex debugging, one can setup `DaVinci` with `graphviz
> > +> ```shell
> > +> SetupDaVinci --use "graphviz v* LCG_Interfaces"
> > +> ```
> > +>
> > +> And add a nice graphical representation of the selection sequence
> > +> ```python
> > +>from SelPy.graph import graph
>
> SelPy is really a top level package or is it a previously defined name?
> Seems too short to be a LHCb top level name
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lhcb/first-analysis-steps/pull/90/files#r31601822>.
>



-- 
Dr. Albert Puig Navarro
Laboratoire de Physique des Hautes Energies
Ecole Polytechnique Fédérale de Lausanne (EPFL)
BSP 614.4 (Cubotron UNIL) CH-1015 Lausanne
EPFL Phone: 021 6939808
CERN Phone: 72518